src/hooks.js

```diff
@@ -61,3 +62,36 @@ export function useSteps(steps) {
     step,
   }
 }
+
+export function usePromise(fn, memoParams, defaultValue) {
+  const [result, setResult] = useState(defaultValue)
+  useEffect(() => {
+    let cancelled = false
+    fn()
+      .then(value => {
+        if (!cancelled) {
+          setResult(value)
+        }
+        return null
+      })
+      .catch(e => console.error('An error occured: ', e))
+    return () => {
+      cancelled = true
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [...memoParams, fn])
+  return result
+}
+
+export function useRepoDetails(baseUrl, detailsUrl) {
+  const fetchDescription = async () => {
+    try {
+      const raw = await fetch(`${baseUrl}${removeStartingSlash(detailsUrl)}`)
+      return raw.text()
+    } catch (e) {
+      console.log('Error fetching decription: ', e)
```

Comment:
We should use `log()` from `utils.js` here: https://github.com/aragon/aragon/blob/master/src/utils.js#L87

Also maybe we could return an empty string?